### ISSUE TITLE
Add LICENSE?

### ISSUE THREAD

**Issue by nboyd** [samuela/torch2jax#8]
Hi,

This is an extremely really cool idea that I'd love to use for an open-source project (translating [boltz-1]( https://github.com/jwohlwend/boltz) to JAX); but to do that you'd need to add a permissive license. Would you mind doing so?

Thanks!

**Comment by samuela** [comment]
Hi @nboyd, thanks for your kind words! I am a fan of your work, and eager to help out boltz-1 any way I can.

I have been meaning to set torch2jax's license to follow something along the lines of https://github.com/samuela/sshenanigans?tab=readme-ov-file#licensing.

In the meantime, I hereby grant you and boltz-1 an MIT license in perpetuity as long as you try running it on HLA-B27:05 at some point :)

**Comment by nboyd** [comment]
Oh thanks! I'm not associated with the boltz team by the way; I wanted to demonstrate your tracing approach on an open-source translation of their model. Unfortunately because I'd like to use that translation later in a commercial application I won't be able to use your code unless it's under a similar license to boltz-1 (e.g. MIT). 

p.s. have you played around with something[like this](https://gist.github.com/nboyd/99a575009d39029e5a8c23e308824429) for randomness? It's a bit inelegant, but it seems to work.

**Comment by samuela** [comment]
Hi @nboyd, gotcha -- I'm confident we can works something out. Shoot me an email? (email in github bio) I'd also love to hear more about what Atomic is up to :)

**Comment by samuela** [comment]
> p.s. have you played around with something[like this](https://gist.github.com/nboyd/99a575009d39029e5a8c23e308824429) for randomness? It's a bit inelegant, but it seems to work.

Yes, this is more or less the designed I had in mind. If possible, I believe it may be preferable to use [this](https://pytorch.org/docs/stable/notes/extending.html#extending-all-torch-api-with-modes) PyTorch interface instead of rolling our own.

**Comment by samuela** [comment]
> p.s. have you played around with something[like this](https://gist.github.com/nboyd/99a575009d39029e5a8c23e308824429) for randomness? It's a bit inelegant, but it seems to work.

Hi @nboyd, inspired by this convo I just completed a fairly extensive rewrite to add support for random ops (dropout, etc) and buffer support (batch norm). It should be pushed to main now. Lmk what you think!

I'll announce it as a new release shortly, but wanted to send it your way for feedback first

**Comment by nboyd** [comment]
very cool, super elegant work! 

p.s. I ended up manually translating Boltz, if you're curious check it out here: [joltz](https://github.com/nboyd/joltz). Transformations like this would be super easy with t2j!
